### PR TITLE
Add spell checking!

### PR DESCRIPTION
**Pull request by vsoch**
How lovely that we would never need to worry about this again! I added this GitHub action this last weekend to a tool called "typos" that I'm digging right now. It works really well! I also renamed the file to "linting" and fixed the extensions to be consistent.

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -291,12 +291,23 @@ nice features to make it less error prone, discussed next.
...
+## URLChecker and Spelling
 
 The [URLschecker](https://github.com/urlstechie/URLs-checker) is a GitHub action
 that @vsoch worked on to contribute retry and some other nice features for the 
 repository here. These features are available as of version 0.1.6 that is used
-in the [workflow](.github/workflows/urlchecker.yml).
+in the [workflow](.github/workflows/urlchecker.yml). In addition, @vsoch found
+a Rust tool called [crate-ci/typos](https://github.com/marketplace/actions/typos-action)
+and contributed an equivalent action so all posts and pages are spell checked.
+If you CI fails, the spelling suggestions will be shown and you can manually
```
Comment (mtbc) [inline README.md:302]:
```suggestion
If your CI fails, the spelling suggestions will be shown and you can manually
```

Comment (vsoch) [reply README.md:302]: I guess the spell checker didn't help me with my bad grammar! :laughing:

**Comment by mtbc** [comment]
How would you suggest testing? Maybe I should open a PR that includes this *and* a spelling mistake and see how helpfully it fails?

**Comment by vsoch** [comment]
I sort of did that here already - if you look at the other changed files those were spelling mistakes that the client found and fixed.

**Comment by vsoch** [comment]
Also I don't think it's our responsibility to test an integration - it has it's own tests! https://github.com/crate-ci/typos

**Comment by mtbc** [comment]
I figured this PR was probably the "passing", I wondered if we had a failing pipeline to compare. It's good it has its own tests but I guess they can't test if it's correctly integrated into this repository so as to have the desired effect. Happy to take your lead though on how eagerly we merge, this looks good to me and I suppose it gets another look anyway when finally your fork gets merged into upstream.

**Comment by vsoch** [comment]
We have so much work to do, we need to get big changes in before we really dig into details. This seems like at tiny detail, and if you are good / approve I say we merge.

**Comment by mtbc** [comment]
I'm amused that if the PR works at all is a tiny detail but no objections from me to merging. :smile:

**Comment by vsoch** [comment]
I don't understand what you mean by "works"

**Comment by mtbc** [comment]
Ha, sorry, I took your comment as referencing mine directly before it, so I misunderstood your meaning, I hadn't thought to connect the DEI question to this PR. So, yes, I figured that, part of testing a PR that introduces spellchecking into a repository is to see it detect a misspelling in that repository, because I supposed that maybe there is something odd about this repository or some slip has been made in referencing or configuring the checker such that, while it abstractly should and can behave as hoped, is not doing in this case, maybe I'm just too used to disappointing surprises.

I could easily be missing some background here. For example, maybe this PR doesn't do quite what I'm guessing, or GitHub things aren't set up for this fork such that this would even be expected to work here, it'll just work for the main USRSE website repository, or maybe you were looking for some other kind of review. I much appreciate your work here and I really don't want to block this or waste your time just because I'm confused. I sure don't require to be in the way of this one, I'd rather help than hinder, so please do move forward as seems best.

**Comment by vsoch** [comment]
> So, yes, I figured that, part of testing a PR that introduces spellchecking into a repository is to see it detect a misspelling in that repository, because I supposed that maybe there is something odd about this repository or some slip has been made in referencing or configuring the checker such that, while it abstractly should and can behave as hoped, is not doing in this case, maybe I'm just too used to disappointing surprises.

The difference I think is that this isn't some new tool I'm developing here, it's an existing GitHub integration. Based on this logic, we would also need to test the github checkout action, the urlchecker action, the pull request action, etc. Testing an integration is not in scope of a repository that uses it. I would say the current run (and spelling fixes that are visible in this PR) are sufficient.

**Comment by mtbc** [comment]
Cool. :+1: You sure have more experience with integrations than I do, I hadn't even realized that crate is a built-in GitHub thing like pull requests, etc., I thought it was pulling in some third-party tool that GitHub was simply triggering. Makes good sense to use it.

**Review by mtbc — APPROVED** [review]

**Comment by vsoch** [comment]
Thanks @mtbc ! And don't worry, we can improve upon this and generally every small detail as we work on the site. I'll merge now to give us a smaller set of issues to talk about tomorrow.

**Pull request merged**